pkg/operator/ceph/file/dependent.go

```diff
@@ -49,3 +70,43 @@ func CephFilesystemDependents(clusterdCtx *clusterd.Context, clusterInfo *client
 
 	return deps, nil
 }
+
+// return subvolume groups that have 1 or more subvolumes present in them
+func subvolumeGroupDependents(clusterdCtx *clusterd.Context, clusterInfo *client.ClusterInfo, filesystem *v1.CephFilesystem) (*dependents.DependentList, error) {
+	baseErr := "failed to get Ceph subvolume groups containing subvolumes"
+
+	deps := dependents.NewDependentList()
+
+	svgs, err := client.ListSubvolumeGroups(clusterdCtx, clusterInfo, filesystem.Name)
+	if err != nil {
+		return deps, errors.Wrap(err, baseErr)
+	}
+
+	errs := []error{}
+	for _, svg := range svgs {
+		svs, err := client.ListSubvolumesInGroup(clusterdCtx, clusterInfo, filesystem.Name, svg.Name)
+		if err != nil {
+			errs = append(errs, errors.Wrapf(err, "failed to list subvolumes in subvolume group %q", svg.Name))
+		}
+
+		if svg.Name == indexSubvolumeGroupName {
+			// TODO: there are two mutually exclusive design options here:
+			//   1. this check is future proof in the event that other metadata subvolumes are added
+			//      to '_index' like 'clone' exists today (not sure how likely this is)
+			//      (this is what is implemented currently)
+			//   2. we can explicitly only ignore the 'clone' subvolume but block if there are
+			//      like if a user deliberately put a subvolume into '_index' (unlikely)
+			logger.Debugf("skipping dependency check for subvolumes in subvolumegroup %q", indexSubvolumeGroupName)
+			continue
+		}
+
+		if len(svs) > 0 {
+			deps.Add(subvolumeGroupDependentType, svg.Name)
+		}
+	}
```

Comment:
@leseb @travisn @Madhu-1 regarding the TODO in this block, what are your thoughts?

Also, to respond to one of Madhu's previous questions, in my testing, the `_deleting` subvolumegroup will be empty once Ceph is done deleting the volumes, so I don't think there is a strong need for Rook to ignore that group. If deletion is blocked on something there, it will eventually be purged. We could still choose to ignore it if it seems better to everyone else. I'm open to suggestions.

Reply:
Ignoring just `_index` keeps it simple and seems reasonable.

Reply:
Sounds good to me, we can ignore `_index` and `_legacy`